DocSum/ui/gradio/docsum_ui_gradio.py

```diff
@@ -91,6 +92,42 @@ def read_video_file(self, file):
         base64_str = self.encode_file_to_base64(file)
         return self.generate_summary(base64_str, document_type="video")
 
+    def is_valid_url(self, url):
+        try:
+            result = urlparse(url)
+            return all([result.scheme, result.netloc])
+        except ValueError:
+            return False
+
+    def read_url(self, url):
+        """Read and process the content of a url.
+
+        Args:
+            url: The url to be read as a document.
+
+        Returns:
+            str: The content of the website or an error message if the url is unsupported.
+        """
+
+        self.page_content = ""
+
+        logger.info(">>> Reading url: %s", url)
+        if self.is_valid_url(url=url):
+            os.environ["no_proxy"] = f"{os.environ.get('no_proxy', '')},{url}".strip(",")
+            try:
+                loader = UnstructuredURLLoader([url])
+                page = loader.load()
+                self.page_content = [content.page_content for content in page][0]
```

Comment:
Why take only the first page_content?

Reply:
there is only one page, just adding [0] to extract the text from the list